frontend/src/metabase/actions/containers/ActionParametersInputForm/utils.ts

```diff
@@ -43,9 +43,10 @@ export const getChangedValues = (
   const changedValues = Object.entries(newValues).filter(
     ([newKey, newValue]) => {
       const oldValue = oldValues[newKey];
+      const hadUnsetValue = oldValue === null || oldValue === undefined;
 
       // don't flag a change when the input changes itself to an empty string
-      if (oldValue === null && newValue === "") {
+      if (hadUnsetValue && newValue === "") {
```

Comment:
@iethree I'm not sure if adding `undefined` handling is the best fix here. Should we rather ensure we use `null` instead of `undefined`? What do you think?